# Imports

In [1]:
import pandas as pd
import zipfile

# Data loading

To unpack our dataset I used zipfile function and after this was used classic function to read data - pd.read_csv. 

In [2]:
with zipfile.ZipFile('bulidata.csv.zip', 'r') as file:
    file.extractall('')
    
df = pd.read_csv('bulidata.csv')
df

Unnamed: 0           MATCH_DATE                       LEAGUE_NAME  \
0              0  2005-08-05 20:30:00  1. Fussball-Bundesliga 2005/2006   
1              1  2005-08-06 15:30:00  1. Fussball-Bundesliga 2005/2006   
2              2  2005-08-06 15:30:00  1. Fussball-Bundesliga 2005/2006   
3              3  2005-08-06 15:30:00  1. Fussball-Bundesliga 2005/2006   
4              4  2005-08-06 15:30:00  1. Fussball-Bundesliga 2005/2006   
...          ...                  ...                               ...   
5503        5505  2023-05-27 15:30:00   1. Fußball-Bundesliga 2022/2023   
5504        5506  2023-05-27 15:30:00   1. Fußball-Bundesliga 2022/2023   
5505        5507  2023-05-27 15:30:00   1. Fußball-Bundesliga 2022/2023   
5506        5508  2023-05-27 15:30:00   1. Fußball-Bundesliga 2022/2023   
5507        5509  2023-05-27 15:30:00   1. Fußball-Bundesliga 2022/2023   

      SEASON LEAGUE  FINISHED   LOCATION  VIEWER      MATCHDAY  MATCHDAY_NR  \
0       2005    bl1      True    München     NaN   1. Spieltag            1   
1       2005    bl1      True       Köln     NaN   1. Spieltag            1   
2       2005    bl1      True   Duisburg     NaN   1. Spieltag            1   
3       2005    bl1      True    Hamburg     NaN   1. Spieltag            1   
4       2005    bl1      True  Wolfsburg     NaN   1. Spieltag            1   
...      ...    ...       ...        ...     ...           ...          ...   
5503    2022    bl1      True       Köln     NaN  34. Spieltag           34   
5504    2022    bl1      True    Leipzig     NaN  34. Spieltag           34   
5505    2022    bl1      True  Stuttgart     NaN  34. Spieltag           34   
5506    2022    bl1      True     Berlin     NaN  34. Spieltag           34   
5507    2022    bl1      True  Wolfsburg     NaN  34. Spieltag           34   

      ...                                          HOME_ICON AWAY_TEAM_ID  \
0     ...                    https://i.imgur.com/jJEsJrj.png           87   
1     ...  https://upload.wikimedia.org/wikipedia/en/thum...           81   
2     ...  https://upload.wikimedia.org/wikipedia/en/c/c8...           16   
3     ...  https://upload.wikimedia.org/wikipedia/commons...           79   
4     ...                    https://i.imgur.com/ucqKV4B.png            7   
...   ...                                                ...          ...   
5503  ...  https://upload.wikimedia.org/wikipedia/en/thum...           40   
5504  ...                    https://i.imgur.com/Rpwsjz1.png            9   
5505  ...                    https://i.imgur.com/v0tkpNx.png          175   
5506  ...  https://assets.dfb.de/uploads/000/018/232/smal...          134   
5507  ...                    https://i.imgur.com/ucqKV4B.png           54   

                AWAY_TEAM_NAME   AWAY_TEAM  \
0     Borussia Mönchengladbach    Gladbach   
1              1. FSV Mainz 05       Mainz   
2                VfB Stuttgart   Stuttgart   
3               1. FC Nürnberg    Nürnberg   
4            Borussia Dortmund         BVB   
...                        ...         ...   
5503         FC Bayern München      Bayern   
5504             FC Schalke 04     Schalke   
5505       TSG 1899 Hoffenheim  Hoffenheim   
5506             Werder Bremen      Bremen   
5507                Hertha BSC      Hertha   

                                              AWAY_ICON GOALS_HOME GOALS_AWAY  \
0                       https://i.imgur.com/KSIk0Eu.png          3          0   
1     https://upload.wikimedia.org/wikipedia/commons...          1          0   
2                       https://i.imgur.com/v0tkpNx.png          1          1   
3     https://upload.wikimedia.org/wikipedia/commons...          3          0   
4     https://upload.wikimedia.org/wikipedia/commons...          2          2   
...                                                 ...        ...        ...   
5503                    https://i.imgur.com/jJEsJrj.png          1          2   
5504  https://upload.wikimedia

Next was used info() function to see information about entries and Dtype of columns. Also I looking for any null values by .isnull().any() and dataset have 2 columns with null values. This is in viewer and away team columns. 

In [3]:
df.info()
print('____________________________')
print(df.isnull().any())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5508 entries, 0 to 5507
Data columns (total 23 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0      5508 non-null   int64  
 1   MATCH_DATE      5508 non-null   object 
 2   LEAGUE_NAME     5508 non-null   object 
 3   SEASON          5508 non-null   int64  
 4   LEAGUE          5508 non-null   object 
 5   FINISHED        5508 non-null   bool   
 6   LOCATION        5508 non-null   object 
 7   VIEWER          1682 non-null   float64
 8   MATCHDAY        5508 non-null   object 
 9   MATCHDAY_NR     5508 non-null   int64  
 10  HOME_TEAM_ID    5508 non-null   int64  
 11  HOME_TEAM_NAME  5508 non-null   object 
 12  HOME_TEAM       5508 non-null   object 
 13  HOME_ICON       5508 non-null   object 
 14  AWAY_TEAM_ID    5508 non-null   int64  
 15  AWAY_TEAM_NAME  5508 non-null   object 
 16  AWAY_TEAM       5474 non-null   object 
 17  AWAY_ICON       5508 non-null   o

Change the capital letters in columns name to lower letters is done only for comfortable of data manipulation.

In [4]:
df.columns = df.columns.str.lower()
df.columns

Index(['unnamed: 0', 'match_date', 'league_name', 'season', 'league',
       'finished', 'location', 'viewer', 'matchday', 'matchday_nr',
       'home_team_id', 'home_team_name', 'home_team', 'home_icon',
       'away_team_id', 'away_team_name', 'away_team', 'away_icon',
       'goals_home', 'goals_away', 'draw', 'win_home', 'win_away'],
      dtype='object')

At first glance can see the unnamed column doesn't agree with automatic indexing (dataset have 5508 entries, but unnamed column shows as if it had 5510). There is no point in looking for an error, can easily remove this column. 

In [ ]:
df.drop(columns = 'U')